ext/opcache/jit/zend_jit.c

```diff
@@ -4629,12 +4629,7 @@ ZEND_EXT_API void zend_jit_unprotect(void)
 #ifdef HAVE_MPROTECT
 	if (!(JIT_G(debug) & (ZEND_JIT_DEBUG_GDB|ZEND_JIT_DEBUG_PERF_DUMP))) {
 		int opts = PROT_READ | PROT_WRITE;
-#ifdef ZTS
-#ifdef HAVE_PTHREAD_JIT_WRITE_PROTECT_NP
-		if (zend_write_protect) {
-			pthread_jit_write_protect_np(0);
-		}
-#endif
+#if defined(ZTS) && (!defined(__APPLE__) || !defined(__aarch64__))
 		opts |= PROT_EXEC;
 #endif
```

Comment:
This disables ``EXEC`` permissions for **all** threads. Therefore other threads are going to crash while this one won't return the permissions back.

According to https://developer.apple.com/documentation/apple-silicon/porting-just-in-time-compilers-to-apple-silicon we should use ``pthread_jit_write_protect_np()`` without ``mprotect()``, but I wasn't able to make it work.
In my opinion the code should be changed in this way https://github.com/php/php-src/pull/13375/files 

It seems ``pthread_jit_write_protect_supported_np()`` somehow returns false (on github actions).
May be we should set ``com.apple.security.cs.allow-jit`` entitlement...
I don't have access to Mac and I have no idea how to do this.

Reply:
I think it needs [`codesign`](https://developer.apple.com/library/archive/documentation/Security/Conceptual/CodeSigningGuide/Procedures/Procedures.html) (and probably a proper apple license).

Reply:
> I think it needs [`codesign`](https://developer.apple.com/library/archive/documentation/Security/Conceptual/CodeSigningGuide/Procedures/Procedures.html) (and probably a proper apple license).

Can you try this?

Reply:
using self-signed identity.

Reply:
will give it a try sometime today definitively.